src/main/java/com/jcraft/jsch/ChannelSftp.java

```diff
@@ -1495,8 +1495,10 @@ public void close() throws IOException {
           rq.cancel(header, buf);
           try {
             _sendCLOSE(handle, header);
+          } catch (IOException e) {
+            throw e;
           } catch (Exception e) {
-            throw new IOException("error");
+            throw new IOException(e.toString(), e);
```

Comment:
I would suggest doing it like this instead:
```
if (e instanceof IOException)
  throw (IOException) e;
throw new IOException(e.toString(), e);
```

Reply:
I copied this particular code block from another part of the same file. 

Sonar proposes to do this the way I did this
https://rules.sonarsource.com/java/RSPEC-1193

In the end the code is equivalent, so if you care about it, I  can also change the pull request.

Reply:
Ok, I see that now, apologies.